internal/ipam/ipam.go

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package ipam
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+)
+
+type Veths struct {
+	HostSide      net.IPNet
+	ContainerSide net.IPNet
+}
+
+// VethIPs returns the IPs for the host side and the PE side
+// for a given pool on the ith node.
+func VethIPs(pool string, index int) (Veths, error) {
+	_, cidr, err := net.ParseCIDR(pool)
+	if err != nil {
+		return Veths{}, fmt.Errorf("failed to parse pool %s: %w", pool, err)
+	}
+	ones, _ := cidr.Mask.Size()
+	peSide, err := cidrElem(pool, ones, 0)
+	if err != nil {
+		return Veths{}, err
+	}
+	hostSide, err := cidrElem(pool, ones, index+1)
+	if err != nil {
+		return Veths{}, err
+	}
+	return Veths{HostSide: *hostSide, ContainerSide: *peSide}, nil
+}
+
+// VTEPIp returns the IP to be used for the local VTEP on the ith node.
+func VTEPIp(pool string, index int) (net.IPNet, error) {
+	ips, err := sliceCIDR(pool, index, 1)
+	if err != nil {
+		return net.IPNet{}, err
+	}
+	if len(ips) != 1 {
+		return net.IPNet{}, fmt.Errorf("vtepIP, expecting 1 ip, got %v", ips)
+	}
```

Comment:
is this needed? (feels like not trusting sliceCIDR)

Reply:
this is mostly because I wanted to ensure I am getting a value here (ie asking for a number out of the range). I will also add a test for this